src/pages/signin/SignInPage.js

```diff
@@ -85,27 +88,44 @@ class SignInPage extends Component {
             && (!this.props.account.validated || this.props.account.forgotPassword)
             && !Permissions.canUsePasswordlessLogins(this.props.betas);
 
+        let welcomeHeader = '';
         let welcomeText = '';
         if (showValidateCodeForm) {
             if (this.props.account.requiresTwoFactorAuth) {
                 // We will only know this after a user signs in successfully, without their 2FA code
+                welcomeHeader = this.props.isSmallScreenWidth ? '' : this.props.translate('welcomeText.welcomeBack');
                 welcomeText = this.props.translate('validateCodeForm.enterAuthenticatorCode');
             } else {
                 const userLogin = Str.removeSMSDomain(lodashGet(this.props, 'credentials.login', ''));
-                welcomeText = this.props.account.validated
-                    ? this.props.translate('welcomeText.welcomeBackEnterMagicCode', {login: userLogin})
-                    : this.props.translate('welcomeText.welcomeEnterMagicCode', {login: userLogin});
+
+                if (this.props.account.validated) {
+                    welcomeHeader = this.props.isSmallScreenWidth ? '' : this.props.translate('welcomeText.welcomeBack');
+                    welcomeText = this.props.isSmallScreenWidth
+                        ? `${this.props.translate('welcomeText.welcomeBack')} ${this.props.translate('welcomeText.welcomeEnterMagicCode', {login: userLogin})}`
+                        : this.props.translate('welcomeText.welcomeEnterMagicCode', {login: userLogin});
+                } else {
+                    welcomeHeader = this.props.isSmallScreenWidth ? '' : this.props.translate('welcomeText.welcome');
+                    welcomeText = this.props.isSmallScreenWidth
+                        ? `${this.props.translate('welcomeText.welcome')} ${this.props.translate('welcomeText.newFaceEnterMagicCode', {login: userLogin})}`
+                        : this.props.translate('welcomeText.newFaceEnterMagicCode', {login: userLogin});
+                }
```

Comment:
I followed the current pattern for the title/body text, but looking at this I feel like this is hard to read, I'll think about a cleaner way to write this.

Essentially,
If on landing page / login form -> Always show header
if on any subsequent form -> Only show header on Large Screens, concat header into body for Small screens